parser/prism/Translator.cc

```diff
@@ -23,6 +23,33 @@ template <typename... TArgs>
 
 template <typename PrismAssignmentNode, typename SorbetAssignmentNode, typename SorbetLHSNode>
 std::unique_ptr<SorbetAssignmentNode> Translator::translateAssignment(pm_node_t *untypedNode) {
+    static_assert(std::is_same_v<SorbetAssignmentNode, parser::Assign>,
+                  "Invalid assignment node type. Must be `parser::Assign`.");
```

Comment:
If `SorbetAssignmentNode` always has to be `parser::Assign`, then there's not need to have it parameterized

```suggestion
template <typename SorbetAssignmentNode, typename SorbetLHSNode>
std::unique_ptr<parser::Assign> Translator::translateAssignment(pm_node_t *untypedNode) {
```